sus/sus.cc

```diff
@@ -0,0 +1,806 @@
+  using sus::iter::Once;
+  using sus::iter::OnceWith;
+  using sus::iter::Peekable;
+  using sus::iter::Repeat;
+  using sus::iter::RepeatWith;
+  using sus::iter::Reverse;
+  using sus::iter::Scan;
+  using sus::iter::SizeHint;
+  using sus::iter::Skip;
+  using sus::iter::SkipWhile;
+  using sus::iter::StepBy;
+  using sus::iter::Successors;
+  using sus::iter::Take;
+  using sus::iter::TakeWhile;
+  using sus::iter::Zip;
+
+  // Functions
+  using sus::iter::begin;
+  using sus::iter::empty;
+  using sus::iter::end;
+  using sus::iter::from_generator;
+  using sus::iter::from_iter;
+  using sus::iter::from_range;
+  using sus::iter::once;
+  using sus::iter::once_with;
+  using sus::iter::repeat;
+  using sus::iter::repeat_with;
+  using sus::iter::successors;
+  using sus::iter::try_from_iter;
+  using sus::iter::zip;
+
+  // Concepts
+  using sus::iter::DoubleEndedIterator;
+  using sus::iter::ExactSizeIterator;
+  using sus::iter::Extend;
+  using sus::iter::FromIterator;
+  using sus::iter::IntoIterator;
+  using sus::iter::IntoIteratorAny;
+  using sus::iter::Iterator;
+  using sus::iter::IteratorAny;
+  using sus::iter::Product;
+  using sus::iter::Sum;
+  using sus::iter::TrustedLen;
+
+  // Type Aliases
+  using sus::iter::IntoIteratorOutputType;
+  using sus::iter::Option;
+
+  // Concept Aliases
+  using sus::iter::Into;
+  using sus::iter::TriviallyRelocatable;
+
+  namespace __private {
+    using sus::iter::__private::IteratorArchetype;
+    using sus::iter::__private::IntoIteratorArchetype;
+    using sus::iter::__private::TrustedLenMarker;
+    using sus::iter::__private::Step;
+  }
+}  // namespace iter
+
+export namespace sus::marker {
+  // Classes
+  using sus::marker::EmptyMarker;
+  using sus::marker::UnsafeFnMarker;
+
+  // Variables
+  using sus::marker::empty;
+  using sus::marker::unsafe_fn;
+}  // namespace marker
+
+export namespace sus::mem {
+  // Classes
+  using sus::mem::NeverValueConstructor;
+
+  // Functions
+  using sus::mem::addressof;
+  using sus::mem::clone;
+  using sus::mem::clone_into;
+  using sus::mem::clone_or_forward;
+  using sus::mem::data_size_of;
+  using sus::mem::forward;
+  using sus::mem::move;
+  using sus::mem::replace;
+  using sus::mem::size_of;
+  using sus::mem::swap;
+  using sus::mem::swap_nonoverlapping;
+  using sus::mem::take;
+  using sus::mem::take_and_destruct;
+  using sus::mem::take_copy_and_destruct;
+
+  // Concepts
+  using sus::mem::Clone;
+  using sus::mem::CloneFrom;
+  using sus::mem::CloneOrRef;
+  using sus::mem::Copy;
+  using sus::mem::CopyOrRef;
+  using sus::mem::CopyOrRefOrVoid;
+  using sus::mem::IsMoveRef;
+  using sus::mem::Move;
+  using sus::mem::MoveOrRef;
+  using sus::mem::MoveOrRefOrVoid;
+  using sus::mem::NeverValueField;
+  using sus::mem::TrivialCopy;
+  using sus::mem::TriviallyRelocatable;
+
+  // Type Aliases
+  using sus::mem::remove_rvalue_reference;
+
+  namespace __private {
+    using sus::mem::__private::HasCloneFromMethod;
```

Comment:
How come private stuff has to be in here? Can it be made reachable without being visible? (https://vector-of-bool.github.io/2019/03/31/modules-2.html)